lib/backend/k8s/resources/watcher.go

```diff
@@ -46,7 +46,9 @@ func newK8sWatcherConverter(
 		cancel:     cancel,
 		resultChan: make(chan api.WatchEvent, resultsBufSize),
 	}
-	go wc.processK8sEvents()
+	if wc.k8sWatch != nil {
```

Comment:
Hm, this one feels a little suspect to me - shouldn't the calling code simply not call this function in this case?

Not sure the full story but it feels like this might belong outside of this function.

Reply:
this one is a long-ish story… we'll sync tomorrow.